include/utility/json_utility.hpp

```diff
@@ -47,29 +47,40 @@ inline std::unordered_map<std::string, functionPtr> funcionMap{
 /**
  * @brief API to read VPD offset from JSON file.
  *
- * @param[in] parsedJson - Parsed JSON file for the system.
- * @param[in] vpdFilePath - path to the VPD file.
+ * @param[in] i_sysCfgJsonObj - Parsed JSON file for the system.
+ * @param[in] i_vpdFilePath - path to the VPD file.
  * @return Offset of VPD in VPD file.
  */
-inline size_t getVPDOffset(const nlohmann::json& parsedJson,
-                           const std::string& vpdFilePath)
+inline size_t getVPDOffset(const nlohmann::json& i_sysCfgJsonObj,
+                           const std::string& i_vpdFilePath)
 {
-    size_t vpdOffset = 0;
-    if (!vpdFilePath.empty())
+    if (i_vpdFilePath.empty())
     {
-        if (parsedJson["frus"].contains(vpdFilePath))
+        return 0;
+    }
+
+    if (i_sysCfgJsonObj["frus"].contains(i_vpdFilePath))
+    {
+        return i_sysCfgJsonObj["frus"][i_vpdFilePath].at(0).value("offset", 0);
+    }
+
+    const nlohmann::json& l_fruList =
+        i_sysCfgJsonObj["frus"].get_ref<const nlohmann::json::object_t&>();
+
+    for (const auto& l_fru : l_fruList.items())
+    {
+        const auto l_fruPath = l_fru.key();
+
+        // check if given path is redundant FRU path
+        if (i_vpdFilePath == i_sysCfgJsonObj["frus"][l_fruPath].at(0).value(
```

Comment:
why we are hardcoding & finding at '0' index only ?
In future, if the fru's order gets changed, then this check won't be able to find the 'redundantEeprom' tag.

Reply:
fru's order never changes. and even if it changes, 0th index will have all the properties which are common for the rest of the subfrus under that eeprom.

Reply:
Can you not use contains for finding redundantEeprom?